src/main/java/school/faang/bjs246209/EncryptionService.java

```diff
@@ -0,0 +1,38 @@
+package school.faang.bjs246209;
+
+public class EncryptionService {
+    private final DroidMessageDecrypt<String, Integer, String> decrypted = (msg, key) -> {
+        System.out.printf("Message: '%s' decrypting with algo no: %s\n", msg, key);
+        return cryptWithAlgo(msg, key, false);
+    };
+
+    public void sendMessage(Droid sender, Droid droid, String message, int encryptionKey) {
+        DroidMessageEncryptor encryptor = (msg, key) -> {
+            System.out.printf("Message: '%s' was encrypted with algo no: %s\n", msg, key);
+            return cryptWithAlgo(msg, key, true);
+        };
+
+        String encryptedMessage = encryptor.apply(message, encryptionKey);
+        System.out.println(sender + " is sending message to "  + droid.name());
+        receiveMessage(sender, droid, encryptedMessage, encryptionKey);
+    }
+
+    public void receiveMessage(Droid sender, Droid droid, String message, int encryptionKey) {
+        String decryptedMessage = decrypted.apply(message, encryptionKey);
+        System.out.println(droid + " have received message '" + decryptedMessage + "' from "  + sender.name());
+    }
+
+    private String cryptWithAlgo(String message, int algo, boolean isEncrypt) {
```

Comment:
https://softwareengineering.stackexchange.com/questions/147977/is-it-wrong-to-use-a-boolean-parameter-to-determine-behavior

https://martinfowler.com/bliki/FlagArgument.html